content/articles/how-to-collect-update-and-organize-python-logs/index.md

```diff
@@ -0,0 +1,337 @@
+author: moris-wanyiri
+date: 2021-12-18T00:00:00-10:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/how-to-collect-update-and-organize-python-logs/hero.png
+    alt: How to collect, update and organize python logs
+---
+
+The implicit logging module in Python is designed to provide your applications with crucial permeability. 
+<!--more-->
+This article explains how to use the module to log all of the data needed, route it to desired destinations, and solidify logs to include more information into Python programs.
+
+We can use an external setup document to set up the Python logging subsystem. We can find the details of the logging setup design in the Python standard library.
+The logging library is divided into four classes: `loggers,` `controllers`, `channels,` and `organizations.`
+
+### Table of Contents
+- [Introduction](#introduction)
+- [Table of Contents](#table-of-contents)
+- [Modifying your logs' need level and objective](#modifying-your-logs-need-level-and-objective)
+- [Make a redid arrangement with different loggers and objections.](#make-a-redid-arrangement-with-different-loggers-and-objections)
+- [Fuse tracebacks and exceptional case managing in your logs](#fuse-tracebacks-and-exceptional-case-managing-in-your-logs)
+- [To work with investigating, design your logs in JSON and merge them](#to-work-with-investigating-design-your-logs-in-json-and-merge-them)
+- [Conclusion](#conclusion)
+- [Further activity reading](#further-activity-reading)
+  
+### Modifying your logs' need level and objective
+The `basicConfig` technique for the logging module is the quickest strategy for planning your loggers' ideal direction. However, the [Python documentation](https://docs.python.org/3.7/library/logging.html#logger-objects) proposes making a logger for every module in your application and arranging a logger for each module utilizing `basicConfig` alone can be annoying.
+
+The three significant measures of `basicConfig` are:
+- **level**: plunging succession of events. The available log ranks are `DEBUG`, `INFO`, `WARNING`, `ERROR`, and `CRITICAL`. The default level is `WARNING`.
+- **controller**: determines where the application will pipe your logs.
+- **design**: messages are logged in the format: `LEVEL>:LOGGER NAME>: MESSAGE>`.
+  
+You could be losing out on truly low logs that can help with underlying cause evaluation, based on how the logging module provides `WARNING` and, more important, level logs as a replacement. 
+
+So instead, use a 'FileHandler' to log single or additional documents on the circle, maybe using a `StreamHandler` or a `SocketHandler` to route logs to the control center or an unessential help over the organization.
+
+#### Benefits of logging documents
+While streaming logs to an outside objective, your application doesn't need to reckon with the possibility of organization-related irregularities. 

```

Comment:
For instance, the statement `while streaming logs to an outside objective...`. I had to pause my reading to first understand what `outside objective` stands for. You can make use of easy language and say `when showing logs in your application...`. Or however you find easy. 

A good exercise would be to ask "how easy would a 5 year old understand what I am saying?"